docs/k8s/auth.md

```diff
@@ -3,4 +3,59 @@ title: "Authentication"
 description: "How to authenticate to FloxHub with Kubernetes on Flox"
 ---
 
-WIP
+Kubernetes on Flox (KoF) allows you to run Flox environments in place or on top of Container images.
+Flox environments are accessed centrally via [FloxHub][floxhub] and managed using the Flox CLI.
+
+In the [introduction][intro] we discussed how annotations are used to instruct KoF which Flox environment to run.
+However, we assumed that the referenced environment was publicly available without authentication.
+If you plan to use private environments, you will have to authenticate KoF to access FloxHub using your user credentials.
+
+To do so, you need to first login to FloxHub using the Flox CLI using [`flox auth login`][flox_auth], if you haven't done so already.
+You then create a new Kubernetes secret:
+
+```bash
+flox auth token | kubectl create secret generic floxhub-token --from-file=floxhub-token=/dev/stdin
+```
+
+Finally, you add a secret volume to your pod specification and mount it to `"/var/run/secrets/flox.dev"` inside your container.
+
+A sample specification is shown below:
+
+```yaml
```

Comment:
For this section is may be easier to read if you remove the comments that are common to the general spec shown in the intro. That way you can use comments to highlight what's new (e.g. the volume/secrets stuff).